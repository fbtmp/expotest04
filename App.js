import App from './src/App';
import {registerRootComponent} from 'expo';

registerRootComponent(App);
export default App;
