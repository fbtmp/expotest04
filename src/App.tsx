import {StatusBar} from 'expo-status-bar';
import React, {useState} from 'react';
import {
  Modal,
  Pressable,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import Constants from 'expo-constants';

import * as pkg from '../package.json';

export default function App() {
  const [modalText, setModalText] = useState('');
  const [modalVisible, setModalVisible] = useState(false);

  const showModal = (text: string) => {
    setModalText(text);
    setModalVisible(true);
  };

  return (
    <View style={styles.container}>
      <Modal
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(!modalVisible);
        }}>
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
            <ScrollView>
              <Text style={styles.modalText}>{modalText}</Text>
            </ScrollView>
            <Pressable
              style={[styles.button]}
              onPress={() => setModalVisible(!modalVisible)}>
              <Text style={styles.textStyle}>Close</Text>
            </Pressable>
          </View>
        </View>
      </Modal>
      <View>
        <Text style={styles.boldText}>
          {pkg.name}@{pkg.version}
        </Text>
      </View>
      <View>
        <Pressable
          style={[styles.button]}
          onPress={() => showModal(JSON.stringify(process.env, null, 2))}>
          <Text style={styles.textStyle}>Show env</Text>
        </Pressable>
        <Pressable
          style={[styles.button]}
          onPress={() =>
            showModal(JSON.stringify(Constants.manifest, null, 2))
          }>
          <Text style={styles.textStyle}>Show manifest</Text>
        </Pressable>
        <Pressable
          style={[styles.button]}
          onPress={() =>
            showModal(JSON.stringify(Constants.manifest?.extra, null, 2))
          }>
          <Text style={styles.textStyle}>Show manifest.extra</Text>
        </Pressable>
      </View>

      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  boldText: {
    color: 'black',
    fontWeight: 'bold',
    margin: 8,
    textAlign: 'center',
  },
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalView: {
    margin: 16,
    backgroundColor: 'white',
    padding: 8,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  button: {
    borderRadius: 4,
    padding: 10,
    margin: 4,
    elevation: 2,
    backgroundColor: '#2196F3',
  },
  textStyle: {
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
  },
  modalText: {
    fontFamily: 'monospace',
    marginBottom: 8,
  },
});
