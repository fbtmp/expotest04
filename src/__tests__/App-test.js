import React from 'react';
import renderer from 'react-test-renderer';
import prettyFormat from 'pretty-format';

import App from '../App';

expect.addSnapshotSerializer({
  serialize(val, config, indentation, depth, refs, printer) {
    const serializer = prettyFormat.plugins.ReactElement;
    val.children.splice(0, 1);
    return serializer.serialize(val, config, indentation, depth, refs, printer);
  },
  test(val) {
    return (
      val &&
      val.type === 'Text' &&
      val.children?.length === 1 &&
      !val.children[0].trim().length
    );
  },
});

describe('<App />', () => {
  it('has 4 children', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(4);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
