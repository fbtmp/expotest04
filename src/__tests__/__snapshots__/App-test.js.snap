// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "elevation": 5,
            "margin": 16,
            "padding": 8,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
          }
        }
      >
        <RCTScrollView>
          <View>
            <Text
              style={
                Object {
                  "fontFamily": "monospace",
                  "marginBottom": 8,
                }
              }
            >
              
            </Text>
          </View>
        </RCTScrollView>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#2196F3",
                "borderRadius": 4,
                "elevation": 2,
                "margin": 4,
                "padding": 10,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#2196F3",
            "borderRadius": 4,
            "elevation": 2,
            "margin": 4,
            "padding": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Show env
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#2196F3",
            "borderRadius": 4,
            "elevation": 2,
            "margin": 4,
            "padding": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Show manifest
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "#2196F3",
            "borderRadius": 4,
            "elevation": 2,
            "margin": 4,
            "padding": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Show manifest.extra
      </Text>
    </View>
  </View>
</View>
`;
